### ISSUE TITLE
build from aur failed

### ISSUE THREAD

**Issue by cubercsl** [ros-noetic-arch/ros-noetic-joint-state-controller#1]
I have tried to install `ros-noetic-desktop-full` from aur, but failed in package `ros-noetic-joint-state-controller`.

OS: Arch Linux x86_64 

BTW, I did not use arch4edu because the python version of those packages seems to be lagging.

Some logs
```
yay -S ros-noetic-desktop-full

...

==> Making package: ros-noetic-joint-state-controller 0.15.0-2 (Sun 20 Dec 2020 03:35:54 AM CST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found ros-noetic-joint-state-controller-0.15.0.tar.gz
==> Validating source files with sha256sums...
    ros-noetic-joint-state-controller-0.15.0.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting ros-noetic-joint-state-controller-0.15.0.tar.gz with bsdtar
==> Sources are ready.
==> Making package: ros-noetic-joint-state-controller 0.15.0-2 (Sun 20 Dec 2020 03:35:57 AM CST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting build()...
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using CATKIN_DEVEL_PREFIX: /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/build/devel
-- Using CMAKE_PREFIX_PATH: /opt/ros/noetic
-- This workspace overlays: /opt/ros/noetic
-- Found PythonInterp: /usr/bin/python (found suitable version "3.9.1", minimum required is "3") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python
-- Using default Python package layout
-- Found PY_em: /usr/lib/python3.9/site-packages/em.py  
-- Using empy: /usr/lib/python3.9/site-packages/em.py
-- Using CATKIN_ENABLE_TESTING: ON
-- Skip enable_testing() when building binary package
-- Using CATKIN_TEST_RESULTS_DIR: /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/build/test_results
-- Forcing gtest/gmock from source, though one was otherwise available.
-- Found gtest sources under '/usr/src/gmock': gtests will be built
-- Found gmock sources under '/usr/src/gmock': gmock will be built
CMake Warning at /usr/src/gmock/CMakeLists.txt:43 (project):
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


CMake Deprecation Warning at /usr/src/gmock/CMakeLists.txt:45 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning at /usr/src/googletest/CMakeLists.txt:54 (project):
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


CMake Deprecation Warning at /usr/src/googletest/CMakeLists.txt:56 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Found PythonInterp: /usr/bin/python (found version "3.9.1") 
-- Found Threads: TRUE  
-- Using Python nosetests: /usr/bin/nosetests-3.9
-- catkin 0.8.8
-- BUILD_SHARED_LIBS is on
-- Configuring done
-- Generating done
-- Build files have been written to: /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/build
Scanning dependencies of target joint_state_controller
[ 50%] Building CXX object CMakeFiles/joint_state_controller.dir/src/joint_state_controller.cpp.o
In file included from /usr/include/boost/smart_ptr/detail/sp_thread_sleep.hpp:22,
                 from /usr/include/boost/smart_ptr/detail/yield_k.hpp:23,
                 from /usr/include/boost/smart_ptr/detail/spinlock_gcc_atomic.hpp:14,
                 from /usr/include/boost/smart_ptr/detail/spinlock.hpp:42,
                 from /usr/include/boost/smart_ptr/detail/spinlock_pool.hpp:25,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:29,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /opt/ros/noetic/include/ros/forwards.h:37,
                 from /opt/ros/noetic/include/ros/node_handle.h:31,
                 from /opt/ros/noetic/include/controller_interface/controller_base.h:35,
                 from /opt/ros/noetic/include/controller_interface/controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/usr/include/boost/bind.hpp:36:1: note: ‘#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.’
   36 | BOOST_PRAGMA_MESSAGE(
      | ^~~~~~~~~~~~~~~~~~~~
In file included from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:39,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:228:21: error: field ‘is_running_’ has incomplete type ‘std::atomic<bool>’
  228 |   std::atomic<bool> is_running_;
      |                     ^~~~~~~~~~~
In file included from /usr/include/c++/10.2.0/bits/shared_ptr_atomic.h:33,
                 from /usr/include/c++/10.2.0/memory:85,
                 from /usr/include/boost/smart_ptr/detail/sp_counted_impl.hpp:35,
                 from /usr/include/boost/smart_ptr/detail/shared_count.hpp:27,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:17,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /opt/ros/noetic/include/ros/forwards.h:37,
                 from /opt/ros/noetic/include/ros/node_handle.h:31,
                 from /opt/ros/noetic/include/controller_interface/controller_base.h:35,
                 from /opt/ros/noetic/include/controller_interface/controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/usr/include/c++/10.2.0/bits/atomic_base.h:152:12: note: declaration of ‘struct std::atomic<bool>’
  152 |     struct atomic;
      |            ^~~~~~
In file included from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:39,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:229:21: error: field ‘keep_running_’ has incomplete type ‘std::atomic<bool>’
  229 |   std::atomic<bool> keep_running_;
      |                     ^~~~~~~~~~~~~
In file included from /usr/include/c++/10.2.0/bits/shared_ptr_atomic.h:33,
                 from /usr/include/c++/10.2.0/memory:85,
                 from /usr/include/boost/smart_ptr/detail/sp_counted_impl.hpp:35,
                 from /usr/include/boost/smart_ptr/detail/shared_count.hpp:27,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:17,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /opt/ros/noetic/include/ros/forwards.h:37,
                 from /opt/ros/noetic/include/ros/node_handle.h:31,
                 from /opt/ros/noetic/include/controller_interface/controller_base.h:35,
                 from /opt/ros/noetic/include/controller_interface/controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/usr/include/c++/10.2.0/bits/atomic_base.h:152:12: note: declaration of ‘struct std::atomic<bool>’
  152 |     struct atomic;
      |            ^~~~~~
In file included from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:39,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:240:20: error: field ‘turn_’ has incomplete type ‘std::atomic<int>’
  240 |   std::atomic<int> turn_;  // Who's turn is it to use msg_?
      |                    ^~~~~
In file included from /usr/include/c++/10.2.0/bits/shared_ptr_atomic.h:33,
                 from /usr/include/c++/10.2.0/memory:85,
                 from /usr/include/boost/smart_ptr/detail/sp_counted_impl.hpp:35,
                 from /usr/include/boost/smart_ptr/detail/shared_count.hpp:27,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:17,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /opt/ros/noetic/include/ros/forwards.h:37,
                 from /opt/ros/noetic/include/ros/node_handle.h:31,
                 from /opt/ros/noetic/include/controller_interface/controller_base.h:35,
                 from /opt/ros/noetic/include/controller_interface/controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:35,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/usr/include/c++/10.2.0/bits/atomic_base.h:152:12: note: declaration of ‘struct std::atomic<int>’
  152 |     struct atomic;
      |            ^~~~~~
In file included from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/include/joint_state_controller/joint_state_controller.h:39,
                 from /home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:35:
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h: In instantiation of ‘realtime_tools::RealtimePublisher<Msg>::RealtimePublisher(const ros::NodeHandle&, const string&, int, bool) [with Msg = sensor_msgs::JointState_<std::allocator<void> >; std::string = std::__cxx11::basic_string<char>]’:
/home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:57:114:   required from here
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:67:99: error: using invalid field ‘realtime_tools::RealtimePublisher<Msg>::is_running_’
   67 |     : topic_(topic), node_(node), is_running_(false), keep_running_(false), turn_(LOOP_NOT_STARTED)
      |                                                                                                   ^
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:67:99: error: using invalid field ‘realtime_tools::RealtimePublisher<Msg>::keep_running_’
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:67:99: error: using invalid field ‘realtime_tools::RealtimePublisher<Msg>::turn_’
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h: In instantiation of ‘bool realtime_tools::RealtimePublisher<Msg>::trylock() [with Msg = sensor_msgs::JointState_<std::allocator<void> >]’:
/home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:84:34:   required from here
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:119:11: error: using invalid field ‘realtime_tools::RealtimePublisher<Msg>::turn_’
  119 |       if (turn_ == REALTIME)
      |           ^~~~~
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h: In instantiation of ‘void realtime_tools::RealtimePublisher<Msg>::unlockAndPublish() [with Msg = sensor_msgs::JointState_<std::allocator<void> >]’:
/home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:97:41:   required from here
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:143:5: error: using invalid field ‘realtime_tools::RealtimePublisher<Msg>::turn_’
  143 |     turn_ = NON_REALTIME;
      |     ^~~~~
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h: In instantiation of ‘void realtime_tools::RealtimePublisher<Msg>::stop() [with Msg = sensor_msgs::JointState_<std::allocator<void> >]’:
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:80:5:   required from ‘realtime_tools::RealtimePublisher<Msg>::~RealtimePublisher() [with Msg = sensor_msgs::JointState_<std::allocator<void> >]’
/usr/include/c++/10.2.0/bits/shared_ptr_base.h:632:8:   required from ‘std::__shared_count<_Lp>::__shared_count(_Ptr) [with _Ptr = realtime_tools::RealtimePublisher<sensor_msgs::JointState_<std::allocator<void> > >*; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/10.2.0/bits/shared_ptr_base.h:639:22:   required from ‘std::__shared_count<_Lp>::__shared_count(_Ptr, std::false_type) [with _Ptr = realtime_tools::RealtimePublisher<sensor_msgs::JointState_<std::allocator<void> > >*; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic; std::false_type = std::integral_constant<bool, false>]’
/usr/include/c++/10.2.0/bits/shared_ptr_base.h:1128:64:   required from ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(_Yp*) [with _Yp = realtime_tools::RealtimePublisher<sensor_msgs::JointState_<std::allocator<void> > >; <template-parameter-2-2> = void; _Tp = realtime_tools::RealtimePublisher<sensor_msgs::JointState_<std::allocator<void> > >; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/10.2.0/bits/shared_ptr_base.h:1309:4:   required from ‘std::__shared_ptr<_Tp, _Lp>::_SafeConv<_Yp> std::__shared_ptr<_Tp, _Lp>::reset(_Yp*) [with _Yp = realtime_tools::RealtimePublisher<sensor_msgs::JointState_<std::allocator<void> > >; _Tp = realtime_tools::RealtimePublisher<sensor_msgs::JointState_<std::allocator<void> > >; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic; std::__shared_ptr<_Tp, _Lp>::_SafeConv<_Yp> = void]’
/home/cubercsl/.cache/yay/ros-noetic-joint-state-controller/src/ros_controllers-0.15.0/joint_state_controller/src/joint_state_controller.cpp:57:115:   required from here
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:103:5: error: using invalid field ‘realtime_tools::RealtimePublisher<Msg>::keep_running_’
  103 |     keep_running_ = false;
      |     ^~~~~~~~~~~~~
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h: In member function ‘bool realtime_tools::RealtimePublisher<Msg>::trylock() [with Msg = sensor_msgs::JointState_<std::allocator<void> >]’:
/opt/ros/noetic/include/realtime_tools/realtime_publisher.h:133:3: warning: control reaches end of non-void function [-Wreturn-type]
  133 |   }
      |   ^
make[2]: *** [CMakeFiles/joint_state_controller.dir/build.make:82: CMakeFiles/joint_state_controller.dir/src/joint_state_controller.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:255: CMakeFiles/joint_state_controller.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
error making: ros-noetic-joint-state-controller
```

**Comment by acxz** [comment]
looks like an upstream issue. Can you make an issue at the upstream repo and post the link back here?

**Comment by acxz** [comment]
looks like they fixed but didn't realize a new version. I went ahead and patched it with 55832b9
